integration-libs/opf/base/core/services/opf-cart-handler.service.ts

```diff
@@ -151,18 +263,55 @@ export class OpfCartHandlerService {
   }
 
   deleteCurrentCart(): Observable<boolean> {
-    return this.activeCartFacade.getActiveCartId().pipe(
-      withLatestFrom(this.userIdService.getUserId()),
-      take(1),
-      tap(([cartId, userId]: [string, string]) => {
-        this.multiCartFacade.deleteCart(cartId, userId);
-      }),
-      switchMap(() =>
-        merge(
-          this.eventService.get(DeleteCartSuccessEvent).pipe(map(() => true)),
-          this.eventService.get(DeleteCartFailEvent).pipe(map(() => false))
-        ).pipe(take(1))
-      )
+    if (!this.cartHandlerState.cartId || !this.cartHandlerState.userId) {
+      return of(false);
+    }
+
+    this.multiCartFacade.deleteCart(
+      this.cartHandlerState.cartId,
+      this.cartHandlerState.userId
     );
+
+    return merge(
+      this.eventService.get(DeleteCartSuccessEvent).pipe(map(() => true)),
+      this.eventService.get(DeleteCartFailEvent).pipe(map(() => false))
+    ).pipe(take(1));
+  }
+
+  removeProductFromOriginalCart(
+    productCode: string,
+    quantity: number
+  ): Observable<boolean> {
+    if (
+      !this.cartHandlerState.previousCartId ||
+      !this.cartHandlerState.userId
+    ) {
+      return of(false);
+    }
+    return this.multiCartFacade
+      .getEntry(this.cartHandlerState.previousCartId, productCode)
```

Comment:
We were previously using 'activeCart' to getEntry, but was getting 'undefined' intermittently.
so, moved to 'multipleCart' to fix it.
I guess root cause was due to race condition where previousCart was not yet loaded as activeCart.